src/couch_replicator.erl

```diff
@@ -129,3 +135,53 @@ cancel_replication(RepId, #user_ctx{name = Name, roles = Roles}) ->
             {error, not_found}
         end
      end.
+
+
+-spec stream_terminal_docs_info(binary(), user_doc_cb(), any()) -> any().
+stream_terminal_docs_info(Db, Cb, UserAcc) ->
+    DDoc = <<"_replicator">>,
+    View = <<"terminal_states">>,
+    QueryCb = fun handle_replicator_doc_query/2,
+    Args = #mrargs{view_type = map, reduce = false},
+    Acc = {Db, Cb, UserAcc},
+    try fabric:query_view(Db, DDoc, View, QueryCb, Acc, Args) of
+    {ok, {Db, Cb, UserAcc1}} ->
+        UserAcc1;
+    {not_found, missing} ->
+        UserAcc
+    catch error:database_does_not_exist ->
+        UserAcc
+    end.
+
+
+% Streaming doesn't save memory much here but it is used to provided a similar
+% API as the terminal docs, which is streaming data from the view
+-spec stream_active_docs_info(user_doc_cb(), any()) -> any().
+stream_active_docs_info(Cb, UserAcc) ->
+    {Replies, _BadNodes} = rpc:multicall(couch_replicator_doc_processor, docs, []),
+    lists:foldl(Cb, UserAcc, lists:concat(Replies)).
```

Comment:
`lists:concat` isn't documented to work for this input (it _does_ work, though). `lists:flatten` seems more appropriate?